openmetadata-service/src/main/java/org/openmetadata/service/apps/scheduler/AppJobFactory.java

```diff
@@ -0,0 +1,81 @@
+package org.openmetadata.service.apps.scheduler;
+
+import static org.openmetadata.service.apps.scheduler.AppScheduler.APP_INFO_KEY;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.entity.app.App;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.search.SearchRepository;
+import org.openmetadata.service.util.JsonUtils;
+import org.quartz.Job;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.simpl.SimpleJobFactory;
+import org.quartz.spi.TriggerFiredBundle;
+
+@Slf4j
+public class AppJobFactory extends SimpleJobFactory {
+  private static final ConcurrentHashMap<String, Job> jobMap = new ConcurrentHashMap<>();
+  private final CollectionDAO collectionDAO;
+  private final SearchRepository searchRepository;
+
+  public AppJobFactory(CollectionDAO collectionDAO, SearchRepository searchRepository) {
+    super();
+    this.collectionDAO = collectionDAO;
+    this.searchRepository = searchRepository;
+  }
+
+  @Override
+  public Job newJob(TriggerFiredBundle bundle, Scheduler scheduler) throws SchedulerException {
+    JobDetail jobDetail = bundle.getJobDetail();
+    JobDataMap jobDataMap = bundle.getJobDetail().getJobDataMap();
+    App jobApp = JsonUtils.readOrConvertValue(jobDataMap.get(APP_INFO_KEY), App.class);
+    Class<? extends Job> jobClass = jobDetail.getJobClass();
+
+    try {
+      if (jobMap.containsKey(jobApp.getClassName())) {
```

Comment:
This is not necessary, it can be done as singleton instead of using map to store the object